examples/templating_tera.rs

```diff
@@ -0,0 +1,44 @@
+#![feature(async_await)]
+
+use tera::{self, compile_templates};
+use tide::{self, App, Context, EndpointResult};
+
+// AppState to pass with context and will hold
+// the interface to the tera rendering engine
+struct AppState {
+    template: tera::Tera,
+}
+
+// Render some data into the 'tera-hello-world.html template in examples/templates directory
+async fn index(ctx: Context<AppState>) -> EndpointResult {
+    // Create the context for the template
+    let mut context = tera::Context::new();
+    context.insert("page_title", "Hello from Tera templating!");
+    context.insert("points", &vec!["point1", "point2"]);
+
+    // Render the variables into the template
+    let s = ctx
+        .state()
+        .template
+        .render("tera-hello-world.html", &context)
+        .expect("Failed to render!");
```

Comment:
Could we perhaps use `?` here instead? -- `.expect` is unlikely to be used for production applications.